test/NuGet.Core.Tests/NuGet.Commands.Test/MinClientVersionTests.cs

```diff
@@ -75,22 +75,12 @@ public async Task MinClientVersion_DependencyVersionTooHighAsync()
 
                 await SimpleTestPackageUtility.CreatePackagesAsync(packageSource.FullName, packageAContext, packageBContext);
 
-                Exception ex = null;
-
                 // Act
                 var command = new RestoreCommand(request);
-
-                try
-                {
-                    var result = await command.ExecuteAsync();
-                }
-                catch (Exception exception)
-                {
-                    ex = exception;
-                }
+                var result = await command.ExecuteAsync();
 
                 // Assert
-                Assert.Contains("The 'packageB 1.0.0' package requires NuGet client version '9.0.0' or above, but the current NuGet version is", ex.Message);
+                Assert.Contains("The 'packageB 1.0.0' package requires NuGet client version '9.0.0' or above, but the current NuGet version is", result.LogMessages.First().Message);
```

Comment:
none of the tests modified in this PR validate that the assets file contains the log message. I'd expect the in-memory results to contain the log because the logger has scope longer than `RestoreCommand`. However, `RestoreCommand` needs to write the assets file, including the logger's messages, and if there's an unhandled exception, there's a decent chance that isn't happening.